scripts/jest/devtools/setupEnv.js

```diff
@@ -57,3 +57,41 @@ global._test_react_version_focus = (range, testName, callback) => {
 global._test_ignore_for_react_version = (testName, callback) => {
   test.skip(testName, callback);
 };
+
+// Most of our tests call jest.resetModules in a beforeEach and the
+// re-require all the React modules. However, the JSX runtime is injected by
+// the compiler, so those bindings don't get updated. This causes warnings
+// logged by the JSX runtime to not have a component stack, because component
+// stack relies on the the secret internals object that lives on the React
+// module, which because of the resetModules call is longer the same one.
+//
+// To workaround this issue, we use a proxy that re-requires the latest
+// JSX Runtime from the require cache on every function invocation.
+//
+// Longer term we should migrate all our tests away from using require() and
+// resetModules, and use import syntax instead so this kind of thing doesn't
+// happen.
+jest.mock('react/jsx-dev-runtime', () => {
+  return new Proxy(
+    {},
+    {
+      get(_, prop) {
+        return jest.requireActual('react/jsx-dev-runtime')[prop];
```

Comment:
Shouldn't we mock the functions exported by the runtime? the jsx bindings may not reevaluated after a reset still? I mean i must be working presumably since your tests are getting stacks but I wonder if this has edge cases that trapping the calls themselves wouldn't have?

Reply:
this effectively does the same thing because this code runs each time any of the functions are called. So even though the bindings themselves never change, this inner code always reads the current module from the cache.

Reply:
oh nvm I see the confusiong... I thought I was already doing what your comment suggests but that's not how I implemented it.

Now I'm confused how why it's working.

Reply:
Ah it works because the compiled output is still a method call. Which is unfortunate but not really our issue, it's Babel's.

I'll update it to proxy the function itself anyway. That's what I had intended to do, I just forget the extra layer and then didn't notice because it worked anyway.

Reply:
cool sounds good